src/main/lcdgui/screens/StepEditorScreen.cpp

```diff
@@ -1348,47 +1349,19 @@ int StepEditorScreen::getYOffset()
 
 void StepEditorScreen::adhocPlayNoteEvent(const std::shared_ptr<mpc::sequencer::NoteOnEvent> &noteEvent)
 {
-    int uniqueEnoughID = playSingleEventCounter++;
+	auto adhoc = std::make_shared<NoteOnEventPlayOnly>();
+	noteEvent->CopyValuesTo(adhoc);
+	const auto bus = track->getBus();
 
-    if (playSingleEventCounter < 0)
+    if (bus != 0)
     {
-        playSingleEventCounter = 0;
-    }
-
-    const auto bus = track->getBus();
-    const auto note = noteEvent->getNote();
-    const auto velo = noteEvent->getVelocity();
-    const auto varType = noteEvent->getVariationType();
-    const auto varValue = noteEvent->getVariationValue();
-
-    if (bus == 0)
-    {
-        // TODO MIDI OUT
+		mpc.getEventHandler()->handle(adhoc, track.get(), bus-1);
```

Comment:
I think we can move it out of the if-clause, remove the if/else, and we'll have MIDI handling.